### PR TITLE
Fix bug - Missing response for newly created rules without UIDs #1047

### PR DESCRIPTION
**Pull request by adimova**
Signed-off-by: Ana Dimova <a.dimova@prosyst.bg>

### CODE REVIEW

File: bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java [hunk 1]
```diff
@@ -1026,10 +1026,11 @@ public String toString() {
     }
 
     @Override
-    public void add(Item element) {
+    public Item add(Item element) {
         if (itemRegistry != null) {
-            itemRegistry.add(element);
+            return itemRegistry.add(element);
         }
+        return element;
```
Comment (maggu2810) [inline bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java]:
```text
      * @throws IllegalStateException
       *             if no ManagedProvider is available
```
Should we return element or throw the IllegalStateException as documented in the API?

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java [hunk 2]
```diff
@@ -107,12 +107,13 @@ public void updated(Provider<E> provider, E oldElement, E element) {
...
 
     @Override
-    public void add(E element) {
+    public E add(E element) {
         if (this.managedProvider != null) {
             this.managedProvider.add(element);
         } else {
             throw new IllegalStateException("ManagedProvider is not available");
         }
+        return element;
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java]: I would prefer to move this line in the if branch, so it is more clear, "return element" is done if a managed provider is available only

Comment (adimova) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java]: Fixed

Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java]: Where? I still see it outside of the if...

File: bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java (hunk 1 above)
Comment (kaikreuzer) [inline bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java]: "if no ManagedProvider is available" -> this is already thrown by the itemRegistry.add() call.
Now we throw an IllegalStateException if the itemRegistry is not available - this is a changed behavior and not mentioned in the JavaDoc, we should not do this.

**Comment by adimova** [comment]
pls see here https://github.com/adimova/smarthome/commit/b4ad014b21e183c88071e5a992092c018f503c93

**Comment by kaikreuzer** [comment]
Now that you pushed the change, I can see it as well :-)

**Comment by adimova** [comment]
Sorry, I missed the second comment from @maggu2810, the fix was only for the first one; I'll fix the second now.

File: bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java (hunk 1 above)
Comment (maggu2810) [inline bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java]: ok

**Comment by maggu2810** [comment]
Ok for me

**Pull request merged**